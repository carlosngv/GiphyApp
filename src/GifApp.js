import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
const GifApp = ({ defaultCategory = [] }) => {

    const [categories, setCategories] = useState(defaultCategory);

    return (
        <>
            <h1 className="title is-1">Gif Search</h1>
            <AddCategory setCategories={setCategories} />
            <hr />
            {
                categories.map((category) => {
                    return (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    );
                })
            }
        </>
    );
}

export default GifApp;