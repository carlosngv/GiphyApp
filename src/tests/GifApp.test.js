import { shallow } from 'enzyme';
import GifApp from '../GifApp';

describe('Tests for GifApp component', () => {
    test('Should show component correctly', () => {
        const wrapper = shallow(<GifApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show category list', () => {
        const categories = ['Itachi', 'Sasuke'];
        const wrapper = shallow(<GifApp defaultCategory={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});