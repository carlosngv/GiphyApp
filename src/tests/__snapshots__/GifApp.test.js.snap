// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifApp component Should show component correctly 1`] = `
<Fragment>
  <h1
    className="title is-1"
  >
    Gif Search
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Tests for GifApp component should show category list 1`] = `
<Fragment>
  <h1
    className="title is-1"
  >
    Gif Search
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Itachi"
    key="Itachi"
  />
  <GifGrid
    category="Sasuke"
    key="Sasuke"
  />
</Fragment>
`;
